src/lib/transition_handler/block_sink.ml

```diff
@@ -33,15 +41,27 @@ type t =
       ; consensus_constants : Consensus.Constants.t
       ; genesis_constants : Genesis_constants.t
       ; constraint_constants : Genesis_constants.Constraint_constants.t
-      ; block_window_duration : Time.Span.t
       }
   | Void
 
 type Structured_log_events.t +=
   | Block_received of { state_hash : State_hash.t; sender : Envelope.Sender.t }
   [@@deriving register_event { msg = "Received a block from $sender" }]
 
-let push sink (`Transition e, `Time_received tm, `Valid_cb cb) =
+let push sink
+    ( (block :
+        [> `Block of Mina_block.t Envelope.Incoming.t
+        | `Header of Mina_block.Header.t Envelope.Incoming.t
+        | `Transition of Mina_block.t Envelope.Incoming.t ] )
+    , `Time_received tm
+    , `Valid_cb cb ) =
+  let e =
+    match block with
+    | `Block b | `Transition b ->
+        b
+    | `Header _ ->
+        failwith "Header not supported"
```

Comment:
I wanted some guidance on what to do with this case, adding in support for headers seemed better in another PR as this is big already/